transaction-pool/src/pool.rs

```diff
@@ -393,6 +393,26 @@ impl<T, S, L> Pool<T, S, L> where
 		self.worst_transactions.iter().next().map(|x| x.transaction.transaction.clone())
 	}
 
+	/// Returns the score of the worst transaction if the pool is almost full.
+	///
+	/// This method can be used to determine what is the minimal required score
+	/// for the replacement transaction. If `None` is returned it means that
+	/// there is still plenty of room in the pool. Otherwise we return
+	/// `Some` with the score of the worst transaction.
+	pub fn minimal_entry_score(&self) -> Option<S::Score> {
+		let threshold = |x: usize| x * 9 / 10;
+		let is_full = {
+			self.by_hash.len() > threshold(self.options.max_count)
+			|| self.mem_usage > threshold(self.options.max_mem_usage)
+		};
+
+		if !is_full {
+			return None
+		}
+
+		self.worst_transactions.iter().next().map(|x| x.score.clone())
```

Comment:
Using `next()` here which looks correct (`worst_transactions` is ordered by ascending `score`). However I noticed: https://github.com/paritytech/parity/blob/0335e927e5d2133ceba4b54579094f1b4d810eac/transaction-pool/src/pool.rs#L277
Is the `next_back()` wrong then? Seems like it would check for the least worst transaction to replace.

Reply:
Great catch, I'll prepare a test and fix that issue with `remove_worst`

Reply:
Actually it's correct in `remove_worst`. `ScoreWithRef` has a custom `Ord` implementation that is ascending if the internal score is descending. So the error is actually here, will make sure to add a correct test to cover this.

Reply:
Looks good. Should this one also change to `next_back()` then?

https://github.com/paritytech/parity/blob/b56200916dfb90149ab22901dac5ef951688974b/transaction-pool/src/pool.rs#L393